_kbarticles/2024-12-09-apt-buildpack-cache-issue.md

```diff
@@ -0,0 +1,30 @@
+---
+layout: post
+title: "Apt buildpack cache issue"
+date: December 9, 2024
+excerpt: Apt buildpack cache issue that can affect application deployment
+---
+
+### What does the buildpack cache do?
+
+The buildpack cache (aka "app cache," or "build artifacts cache") is a per-application cache that is stored in the blob (binary large object) store and recovered when an existing application is restaged (regardless of application code or buildpack changes).
+
+Different buildpacks use this cache for different things. [System buildpacks]({% link _docs/deployment/frameworks.md %}#supported-languages-and-frameworks) are cached on the VM hosting the application container and then mounted as read-only volumes into each staging container. If your application references a custom buildpack or buildpack version via a URLs (to a git repo or a zip file), those buildpacks are downloaded every time application staging occurs.
+
+### The potential issue with the apt buildpack
+
+The potential issue involves the `/` directory disk allocation. The apt buildpack uses `/tmp` like other buildpacks. The `/` directory is allocated approximately 9% of the disk allocation. For most buildpacks, this is sufficient as all the possible packages that they could be installing is known with the only variable being the application code. The apt buildpack however, can install a vast number of large packages, suchas  the AWS CLI. Depending on how many packages are installed and their individual size, the allotted 9% of disk space may not be sufficient.
```

Comment:
```suggestion
The potential issue with the `apt-buildpack` cache involves the `/` directory disk allocation. The `apt-buildpack` uses `/tmp` for its cache like other buildpacks. In all buildpacks, the `/` directory is allocated approximately 9% of the disk allocation, meaning that the size of the `/tmp` subdirectory cannot exceed that disk allocation. For most buildpacks, this disk allocation is sufficient as all the possible packages that they could be installing is known with the only variable being the application code. However, the `apt-buildpack` can install an unknown number of packages of variable size, such as the AWS CLI. Depending on how many packages are installed and their individual size, the allotted 9% of disk space may not be sufficient, causing application staging to fail.
```

Do we have an example of the error that customers might see?

Reply:
`stdout: No space left on device. /usr/bin/tar: /tmp/output-cache: Wrote only 2048 of 10240 bytes`

Reply:
Could we add that to this documenation?

Reply:
There's another example on this ticket as well: https://cloud-gov-new.zendesk.com/agent/tickets/9680.